agent/engine/engine_integ_test.go

```diff
@@ -284,3 +283,29 @@ func TestStartStopWithCredentials(t *testing.T) {
 	_, ok := credentialsManager.GetTaskCredentials(credentialsIDIntegTest)
 	assert.False(t, ok, "Credentials not removed from credentials manager for stopped task")
 }
+
+func TestContainerHealthCheck(t *testing.T) {
+	taskEngine, done, _ := setupWithDefaultConfig(t)
+	defer done()
+
+	stateChangeEvents := taskEngine.StateChangeEvents()
+
+	taskArn := "testTaskWithHealthCheck"
+	testTask := createTestHealthCheckTask(taskArn)
+
+	go taskEngine.AddTask(testTask)
+
+	verifyContainerRunningStateChange(t, taskEngine)
+	verifyTaskIsRunning(stateChangeEvents, testTask)
+
+	time.Sleep(healthCheckWaitTime)
+	healthStatus := testTask.Containers[0].GetHealthStatus()
+	assert.Equal(t, "HEALTHY", healthStatus.Status.BackendStatus(), "container health status is not HEALTHY")
+
+	taskUpdate := createTestTask(taskArn)
+	taskUpdate.SetDesiredStatus(api.TaskStopped)
+	go taskEngine.AddTask(taskUpdate)
+
+	verifyContainerStoppedStateChange(t, taskEngine)
+	verifyTaskIsStopped(stateChangeEvents, testTask)
```

Comment:
nit: since the `verifyTaskIs*` functions are slated for deprecation, should we tag the function name to reflect that in a separate commit? might prevent use of `verifyTaskIs*` slipping through reviews. @aaithal, what do you think?

Reply:
I am not sure if they should be deprecated since they also do a check for the right task's state change events by verifying the ARN. We should probably decide on a standard for using either the verify[Container/Task][Started/Stopped]StateChange or verifyTaskIs* methods, or keep the verifyTaskIs* and create similar ones for container state change events too.